rdt/hyper_transformer.py

```diff
@@ -225,23 +225,32 @@ def update_transformers_by_sdtype(self, sdtype, transformer):
             transformer (rdt.transformers.BaseTransformer):
                 Transformer class or instance to be used for the given ``sdtype``.
         """
+        if self._fitted:
+            warnings.warn(self._REFIT_MESSAGE)
+
         if not self.field_sdtypes:
             raise Error(
                 'Nothing to update. Use the `detect_initial_config` method to '
                 'pre-populate all the sdtypes and transformers from your dataset.'
             )
 
+        if sdtype not in self._get_supported_sdtypes():
```

Comment:
I don't think this handles RDT-plus, so even if it is installed, it will still raise an error.

Reply:
Looks like we need to do 2 things:
1. Update the private transformers to use the correct naming
2. Clear the cache when searching transformers
I'll add the second to this PR